Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h

```diff
@@ -1290,7 +1290,17 @@ namespace JSC {
         Vector<std::pair<FunctionMetadataNode*, FunctionVariableType>> m_functionsToInitialize;
         bool m_needToInitializeArguments { false };
         RestParameterNode* m_restParameter { nullptr };
-        
+
+        struct {
+            bool needTryCatch { false };
+            FunctionNode* function { nullptr };
+            RefPtr<Label> catchStartLabel { nullptr };
+            RefPtr<RegisterID> thrownValue { nullptr };
+        } m_asyncFuncParametersTryCatchInfo;
```

Comment:
I think, probably, instead of always having this and checking `needTryCatch`, I would like to have `std::optional<AsyncFuncParametersTryCatchInfo> m_asyncFuncParametersTryCatchInfo`.